### PR TITLE
Add semaphore support

### PR DESCRIPTION
**Pull request by benjaminwood**
Adds support for Semaphore.

This one could use close review to ensure I didn't mess anything up in the refactor.

Here's an example of semaphore build with this configuration: https://selectiveci.semaphoreci.com/jobs/1a5a654c-00c4-49a3-b4ca-4498c499a5bc

### CODE REVIEW

**Comment by natevick** [comment]
👀

**Comment by benjaminwood** [comment]
Fix for the spec failure is in #44. I'll merge main once that PR is in.

File: lib/bin/build_env.sh [hunk 1]
```diff
@@ -3,48 +3,48 @@
...
-  run_attempt=$SELECTIVE_RUN_ATTEMPT
-  runner_id=$SELECTIVE_RUNNER_ID
+  branch=$CIRCLE_BRANCH
+  actor=${CIRCLE_USERNAME:-$CIRCLE_PR_USERNAME}
+  sha=$CIRCLE_SHA1
+  run_attempt=$CIRCLE_BUILD_NUM
+  runner_id=$CIRCLE_NODE_INDEX
+elif [ -n "$SEMAPHORE" ]; then
+  platform=semaphore
+  branch=${SEMAPHORE_GIT_PR_BRANCH:-$SEMAPHORE_GIT_BRANCH}
```
Comment (benjaminwood) [inline lib/bin/build_env.sh:22]: Semaphore is weird

> In builds triggered by a Pull Request, the value of the SEMAPHORE_GIT_BRANCH is the name of the git branch targeted by the Pull Request.

https://docs.semaphoreci.com/ci-cd-environment/environment-variables/#semaphore_git_branch

And

> The value of the SEMAPHORE_GIT_PR_BRANCH environment variable is the name of the git branch from which the Pull Request originated.

https://docs.semaphoreci.com/ci-cd-environment/environment-variables/#semaphore_git_pr_branch

**Review by natevick — APPROVED** [review]
Sweet

**Pull request merged**